astropy/coordinates/sky_coordinate_parsers.py

```diff
@@ -91,55 +91,49 @@ def _get_frame_without_data(args, kwargs):
     # The first place to check: the frame could be specified explicitly
     frame = kwargs.pop("frame", None)
 
-    if frame is not None:
-        # Here the frame was explicitly passed in as a keyword argument.
-
-        # If the frame is an instance or SkyCoord, we extract the attributes
-        # and split the instance into the frame class and an attributes dict
-
-        if isinstance(frame, SkyCoord):
-            # If the frame was passed as a SkyCoord, we also want to preserve
-            # any extra attributes (e.g., obstime) if they are not already
-            # specified in the kwargs. We preserve these extra attributes by
-            # adding them to the kwargs dict:
-            for attr in frame._extra_frameattr_names:
-                if attr in kwargs and np.any(getattr(frame, attr) != kwargs[attr]):
-                    # This SkyCoord attribute passed in with the frame= object
-                    # conflicts with an attribute passed in directly to the
-                    # SkyCoord initializer as a kwarg:
-                    raise ValueError(
-                        _conflict_err_msg.format(
-                            attr, getattr(frame, attr), kwargs[attr], "SkyCoord"
-                        )
+    if isinstance(frame, SkyCoord):
+        # If the frame was passed as a SkyCoord, we also want to preserve
+        # any extra attributes (e.g., obstime) if they are not already
+        # specified in the kwargs. We preserve these extra attributes by
+        # adding them to the kwargs dict:
+        for attr in frame._extra_frameattr_names:
```

Comment:
Could avoid duplicating the `getattr` (and perhaps clarify the logic) by
```
value = getattr(frame, attr)
if attr not in kwargs:
    kwargs[attr] = value
elif np.any(value != kwargs[attr]):
    raise ValueError(...)
```

Reply:
Applying this suggestion indeed made the code less verbose.